modules/afprog/afprog.c

```diff
@@ -329,6 +382,23 @@ afprogram_dd_exit(pid_t pid, int status, gpointer s)
 }
 
 static gboolean
+afprogram_dd_restore_reload_info(AFProgramDestDriver * const self, GlobalConfig * const cfg)
+{
+  AFProgramReloadInfo * const restored_info = (AFProgramReloadInfo *)cfg_persist_config_fetch(cfg, afprogram_dd_format_persist_name(self));
+
+  if (restored_info)
+    {
+      self->pid    = restored_info->pid;
+      self->writer = restored_info->writer;
+
+      child_manager_register(self->pid, afprogram_dd_exit, log_pipe_ref(&self->super.super.super), (GDestroyNotify)log_pipe_unref);
+      g_free(restored_info);
```

Comment:
we should really clear it up whose responsibility it is to register the pid to the child manager. from time-to-time this repsonsibility seems to shift, and I can't follow that clearly. Maybe it's just me though, perhaps you could explain (with a couple of comments that is), which object is responsible for taking care of killing the pid.